src/docgen/view/documentation.ts

```diff
@@ -1,121 +1,227 @@
+          transpile = new TypeScriptTranspile();
+          break;
+        default:
+          throw new Error(`Unsupported language: ${options?.language}`);
+      }
+      const assembly = await createAssembly(assemblyName, workdir, options?.loose ?? true, language);
+      return new Documentation(assembly, transpile);
+    });
+
+  }
+
+  private constructor(
+    private readonly assembly: reflect.Assembly,
+    private readonly transpile: Transpile) {
   }
 
   /**
    * Generate markdown.
    */
-  public render(): Markdown {
+  public render(options?: RenderOptions): Markdown {
+    const submodule = options?.submodule ? this.findSubmodule(this.assembly, options.submodule) : undefined;
     const documentation = new Markdown();
 
-    if (this.includeReadme) {
-      const readme = new Readme(this.transpile, this.assembly, this.submodule);
+    if (options?.readme ?? true) {
+      const readme = new Readme(this.transpile, this.assembly, submodule);
       documentation.section(readme.render());
     }
-    if (this.includeApiReference) {
-      const apiReference = new ApiReference(
-        this.transpile,
-        this.assembly,
-        this.submodule,
-      );
+
+    if (options?.apiReference ?? true) {
+      const apiReference = new ApiReference(this.transpile, this.assembly, submodule);
       documentation.section(apiReference.render());
     }
 
     return documentation;
   }
 
   /**
-   * Lookup a submodule by a submodule name given by the caller.
+   * Lookup a submodule by a submodule name.
    */
-  private findSubmodule(submoduleName: string): reflect.Submodule {
-    const submodules = this.assembly.submodules.filter(
-      (s) => s.name === submoduleName,
+  private findSubmodule(assembly: reflect.Assembly, submodule?: string): reflect.Submodule {
+    const submodules = assembly.submodules.filter(
+      (s) => s.name === submodule,
     );
 
     if (submodules.length === 0) {
-      throw new Error(`Submodule ${submoduleName} not found`);
+      throw new Error(`Submodule ${submodule} not found`);
     }
 
     if (submodules.length > 1) {
-      throw new Error(`Found multiple submodules with name: ${submoduleName}`);
+      throw new Error(`Found multiple submodules with name: ${submodule}`);
     }
 
     return submodules[0];
   }
 }
+
+async function withTempDir<T>(work: (workdir: string) => Promise<T>): Promise<T> {
+  const workdir = await fs.mkdtemp(path.join(os.tmpdir(), path.sep));
+  const cwd = process.cwd();
+  try {
+    process.chdir(workdir);
+    // wait for the work to be completed before
+    // we cleanup the work environment.
+    return await work(workdir);
+  } finally {
+    process.chdir(cwd);
+    await fs.remove(workdir);
+  }
+}
+
+async function createAssembly(name: string, tsDir: string, loose: boolean, language?: TargetLanguage): Promise<reflect.Assembly> {
+  const ts = new reflect.TypeSystem();
+  for (let dotJsii of await glob.promise(`${tsDir}/**/.jsii`)) {
+    if (language) {
```

Comment:
TypeScript should be skipped probably?

Reply:
It is skipped, this is exactly the scenario where the target language is undefined, again stemmed from the fact typescript isn't really a target language in rosetta.